app/main.py

```
@@ -1,5 +1,86 @@
+from typing import List, Optional
+
+
 class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None
+    ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords is not None else [0, 0]
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: Optional[List[int]] = None
+    ) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
+        else:
+            if len(coords) == 2:
+                coords.append(0)
```

Here, you're appending a `0` to `coords` if it has a length of 2. This makes sense for a `FlyingRobot`, since it needs three coordinates. However, this could potentially lead to unexpected behavior if `coords` is a list with more than two elements. If `coords` has three elements, the third element will be ignored and replaced with `0`. If `coords` has more than three elements, the extra elements will be ignored. It might be better to raise an error if `coords` does not have exactly two or three elements.